### PR TITLE
sclang: AnalogOut and DigitalOut return 0.0

### LINKED ISSUE #61: Bela UGens do not interact nicely with `play`
@apmcpherson had found an issue with one of the digital UGens I think, where having it last in a `play` statement would occasionally cause noisy outputs (probably because the output buffer was not being filled with zeros).

Now I have this:
```
s.waitForBoot{
    "Server Booted".postln;
    {
        var rate = AnalogIn.ar(0).exprange(0.3, 20);
        var amp = AnalogIn.ar(1); // returns a value from 0-1
        rate.poll(1);
        amp.poll(1);
        AnalogOut.ar(0, SinOsc.ar(rate).range(0.0, amp)); // send to Analog Output 0
    }.play;
}
```
and this floods the terminal output with
```
UGen(AnalogIn): 0.000106812
```
(they are about 50 per second: the trigger input to the Poll receives a trigger every 873 or 889 samples), while the  lines from the `LinExp` (generated by `rate.poll`)
```
UGen(LinExp): 0.303403
```
seem to come in at about the right frequency.

Funny enough, if you comment out, e.g.: `rate.poll(1);`, thus leaving only the AnalogIn Poll UGen, this one only prints its output only once at the beginning (probably when the constructor calls `Poll_next_kk`).

Explicitly adding a UGen which produces a valid audio-rate output as the last command of the `play` block fixes everything, e.g.:
```
s.waitForBoot{
    "Server Booted".postln;
    {
        var rate = AnalogIn.ar(0).exprange(0.3, 20);
        var amp = AnalogIn.ar(1); // returns a value from 0-1
        rate.poll(1);
        amp.poll(1);
        AnalogOut.ar(0, SinOsc.ar(rate).range(0.0, amp)); // send to Analog Output 0
        Out.ar(0,
            SinOsc.ar(100, 0, 0.1 )
        )
    }.play;
}
```

@sensestage: is this expected behaviour with `play` or is there something wrong in the Bela UGens?

Should we just avoid `play` altogether and do something like this (which works fine) instead?
```
s.waitForBoot{
SynthDef("AnalogIn",{ arg out=0;
    var rate = AnalogIn.ar(0).exprange(0.3, 20);
    var amp = AnalogIn.ar(1); // returns a value from 0-1
    rate.poll(1);
    amp.poll(1);
    AnalogOut.ar(0, SinOsc.ar(rate).range(0.0, amp)); // send to Analog Output 0
}).send(s);
s.sync;
a = Synth.new("AnalogIn", target: s).postln;
}
```

### PR DESCRIPTION
**Pull request by elgiano**
Purpose and Motivation
----------------------
Fixes #61.
For UGens without any output, such as AnalogOut and DigitalOut, returning 0.0 prevents the creation of a wrapping Out UGen, which in turn was seen messing with poll rates in issue #61.
This technique was copied from other outputless UGens, such as SendTrig.

Types of changes
----------------

- Bug fix (non-breaking change which fixes an issue)

Checklist
---------

- [x] All tests are passing
- [x] This PR is ready for review

Remaining Work
--------------

### CODE REVIEW

File: SCClassLibrary/Common/Audio/bela/BELAUGens.sc [hunk 1]
```diff
@@ -68,10 +70,12 @@ DigitalIn : UGen {
  */
 DigitalOut : UGen {
     *ar { arg digitalPin = 0, output=0, writeMode=0, mul=1.0, add=0.0;
-        ^this.multiNew('audio', digitalPin, output, writeMode ).madd(mul,add)
+        this.multiNew('audio', digitalPin, output, writeMode ).madd(mul,add);
+		^0.0;
```
Comment (giuliomoro) [inline SCClassLibrary/Common/Audio/bela/BELAUGens.sc]: something weird with indentation. You should use `tools/clang-format.py` to lint your files prior to commit

**Review by giuliomoro — COMMENTED** [review]
No idea what this does :)  (because of my ignorance in Sc)

Fix the indentation and we should be fine! Is this needed also for the analog and digital ins ?

File: SCClassLibrary/Common/Audio/bela/BELAUGens.sc (hunk 1 above)
Comment (elgiano) [reply SCClassLibrary/Common/Audio/bela/BELAUGens.sc]: Sorry, the problem is that clang-format doesn't format .sc files, and these files are all in spaces, while all .sc files in main SuperCollider are in tabs. I'll put those lines in spaces as well, then maybe we change all of them to tabs at once before merge with main sc?

**Comment by elgiano** [comment]
Analog and Digital In have outputs, so it should be all right as they are :)

File: SCClassLibrary/Common/Audio/bela/BELAUGens.sc (hunk 1 above)
Comment (giuliomoro) [reply SCClassLibrary/Common/Audio/bela/BELAUGens.sc]: yeah we made a mess the moment we started editing these ... what I do in these cases is I try to keep the convention of the surrounding code. I thought the linter would take care of these, but clearly it doesn't, thanks for letting me know and fixing it.

**Comment by elgiano** [comment]
I see that DigitalIn/Out are not "supposed" to have modulatable pin index... shall I change this, make it modulatable and put a check there, or shall I leave it as it is?

**Comment by giuliomoro** [comment]
I don't think that's needed... Or at least no one asked for it yet!

**Comment by elgiano** [comment]
Ops, sorry I posted in the wrong PR, I was referring to your request in #76, I'll continue back there

**Pull request merged**